pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -167,7 +167,7 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 		if resourceContains(configSource.SubscribedResources, meshconfig.Resource_SERVICE_REGISTRY) {
 
 			//TODO(Nino-K): https://github.com/istio/istio/issues/16976
-			args.Service.Registries = []string{string(serviceregistry.MCPRegistry)}
+			args.Service.Registries = []string{string(serviceregistry.MCP)}
```

Comment:
I don't think it's needed, I have made a [PR](https://github.com/istio/istio/pull/19140) to remove it and fix the native test that was failing because of this case.